addons/mail/models/mail_thread.py

```diff
@@ -520,15 +520,23 @@ def _get_tracked_fields(self, updated_fields):
         """ Return a structure of tracked fields for the current model.
             :param list updated_fields: modified field names
             :return dict: a dict mapping field name to description, containing on_change fields
+            :track_sequence add possibilities to sequence fields
         """
         tracked_fields = []
+        res = {}
         for name, field in self._fields.items():
             if getattr(field, 'track_visibility', False):
-                tracked_fields.append(name)
+                tracked_fields.append([name, getattr(field,'track_sequence', 100)])
 
-        if tracked_fields:
-            return self.fields_get(tracked_fields)
-        return {}
+        tracked_fields_ordered = sorted(tracked_fields, key=lambda colonnes: colonnes[1])
+
+        if not tracked_fields_ordered:
+            return res
+
+        for name, field in tracked_fields_ordered:
```

Comment:
It seems you rely on dict giving back keys in the same order you added them + tracking being displayed using ids ascending order. I don't think it is a good idea as it will probably break without noticing someone.

I would instead have a sequence field on tracking model, copying the sequence used in the model fields definition. Order of the tracking will be sequence asc, meaning we can always rely on database value to rebuild the message tracking values display.

Finally I would really love having tests for that feature in test_mail to ensure it is a standardized behavior.

Thanks !